app/forms/declaration/CommodityMeasure.scala

```diff
@@ -28,13 +28,15 @@ object CommodityMeasure {
   val commodityFormId = "CommodityMeasure"
 
   val mapping = Forms.mapping(
-    "supplementaryUnits" -> optional(text().verifying("supplementary.commodityMeasure.supplementaryUnits.error", validateDecimal(16)(2))),
+    "supplementaryUnits" -> optional(
+      text().verifying("supplementary.commodityMeasure.supplementaryUnits.error", validateDecimal(16)(6) and containsNotOnlyZeros)
+    ),
     "netMass" -> text
       .verifying("supplementary.commodityMeasure.netMass.empty", nonEmpty)
-      .verifying("supplementary.commodityMeasure.netMass.error", isEmpty or validateDecimal(11)(3)),
+      .verifying("supplementary.commodityMeasure.netMass.error", isEmpty or validateDecimal(16)(6) and containsNotOnlyZeros),
     "grossMass" -> text
       .verifying("supplementary.commodityMeasure.grossMass.empty", nonEmpty)
-      .verifying("supplementary.commodityMeasure.grossMass.error", isEmpty or validateDecimal(16)(2))
+      .verifying("supplementary.commodityMeasure.grossMass.error", isEmpty or validateDecimal(16)(6) and containsNotOnlyZeros)
```

Comment:
Why you changed `validateDecimal` values?

Reply:
we validated with BA and found that error message and accordingly implementation was not correct . we should allow user to enter 6 digits after decimal instead of 2. 

is it okay?

Reply:
Yes, but if you can please ask BA to update a ticket with those information. It can be necessary for testing purposes.

Reply:
Done